proxy/src/coco.rs

```diff
@@ -1,649 +1,457 @@
-
-    let tags = platinum_browser
-        .list_tags()
-        .expect("unable to get list of tags")
-        .iter()
-        .map(|t| format!("+refs/tags/{}", t.name()))
-        .collect::<Vec<String>>();
-
-    {
-        let branches = platinum_repo.branches(Some(git2::BranchType::Remote))?;
-
-        for branch in branches {
-            let (branch, _branch_type) = branch?;
-            let name = &branch
-                .name()
-                .expect("unable to get branch name")
-                .expect("branch not present")
-                .get(7..)
-                .expect("unable to extract branch name");
-            let oid = branch.get().target().expect("can't find OID");
-            let commit = platinum_repo.find_commit(oid)?;
-
-            if *name != "master" {
-                platinum_repo.branch(name, &commit, false)?;
-            }
-        }
+#[async_trait]
+impl entity::Resolver<user::User> for NoopUserResolver {
+    async fn resolve(&self, _uri: &RadUrn) -> Result<user::User, entity::Error> {
+        todo!()
     }
 
-    // Init as rad project.
-    let (id, meta) = init_project(
-        librad_paths,
-        platinum_into.to_str().expect("unable to get path"),
-        name,
-        description,
-        default_branch,
-    )?;
-    let mut rad_remote = platinum_repo.find_remote("rad")?;
-
-    // Push all tags to rad remote.
-    rad_remote.push(&tags.iter().map(String::as_str).collect::<Vec<_>>(), None)?;
-    // Push dev branch.
-    rad_remote.push(&["+refs/heads/dev"], None)?;
-
-    Ok((id, meta))
-}
-
-/// Creates a small set of projects in [`Paths`].
-///
-/// # Errors
-///
-/// Will error if filesystem access is not granted or broken for the configured
-/// [`librad::paths::Paths`].
-pub fn setup_fixtures(librad_paths: &Paths, root: &str) -> Result<(), error::Error> {
-    let infos = vec![
-        ("monokel", "A looking glass into the future", "master"),
-        (
-            "Monadic",
-            "Open source organization of amazing things.",
-            "master",
-        ),
-        (
-            "open source coin",
-            "Research for the sustainability of the open source community.",
-            "master",
-        ),
-        (
-            "radicle",
-            "Decentralized open source collaboration",
-            "master",
-        ),
-    ];
-
-    for info in infos {
-        let path = format!("{}/{}/{}", root, "repos", info.0);
-        std::fs::create_dir_all(path.clone())?;
-
-        init_repo(path.clone())?;
-        init_project(librad_paths, &path, info.0, info.1, info.2)?;
+    async fn resolve_revision(
+        &self,
+        _uri: &RadUrn,
+        _revision: u64,
+    ) -> Result<user::User, entity::Error> {
+        todo!()
     }
+}
 
-    Ok(())
+/// Constructs a fake user to be used as an owner of projects until we have more permanent key and
+/// user management.
+pub fn fake_owner() -> user::User {
```

Comment:
Another organisational thing: it'd probably make sense to have `mock` module for putting this and the `Noop` stuff into.